### PR TITLE
Remove txt2tags dependency (use markdown instead)

### PR DESCRIPTION
**Pull request by PeterPetrik**
follow up from discussion on https://github.com/qgis/QGIS/pull/35914

`pandoc --table-of-contents --toc-depth=1 -s -o doc/NEWS.html NEWS.md --metadata title="QGIS News"`
~~pandoc -o doc/INSTALL.html INSTALL.md~~

Need to update these scripts
- [x] ./scripts/release.pl
- [x] ./scripts/update_news.pl
- [x] ./scripts/scandeps.pl

Result:

- https://github.com/PeterPetrik/QGIS/blob/remove_txt2txt/INSTALL.md
- https://github.com/PeterPetrik/QGIS/blob/remove_txt2txt/NEWS.md
- ~~https://github.com/PeterPetrik/QGIS/blob/remove_txt2txt/doc/INSTALL.html~~
- https://github.com/PeterPetrik/QGIS/blob/remove_txt2txt/doc/NEWS.html

### CODE REVIEW

**Comment by PeterPetrik** [comment]
cc @timlinux

**Comment by m-kuhn** [comment]
Good move !

Do we still want/need HTML in the source code?

**Comment by PeterPetrik** [comment]
> Do we still want/need HTML in the source code?

I do not know, it is packaged in some release scripts so I hope @jef-n can say if we need them? Also it looks like that the NEWS is appended automatically, so I am not sure how to do it in this new format...

**Review by Codym48 — COMMENTED** [review]
This is great! I suggested some hierarchical changes in the Windows workflow, but really appreciate you making the time to convert this to .md!

**Comment by nyalldawson** [comment]
Great move!

**Comment by PeterPetrik** [comment]
@jef-n removed INSTALL.html and modified those 3 perl scripts with the new stuff. 

Unfortunately I never used them , and Perl is not my daily bread, so I am wondering if you can take a look if you are happy with the new implementation?

**Comment by PeterPetrik** [comment]
NEWS.html is used in the QGIS desktop panel so we need to maintain it

**Comment by m-kuhn** [comment]
> NEWS.html is used in the QGIS desktop panel so we need to maintain it

Ideally this would be created on build (and not stored in git), but this is nowhere close a blocker for this pull request.

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,1554 @@
+QGIS
+Building QGIS from source - step by step
+
+   1. [Introduction](#1-introduction)
+   2. [Overview](#2-overview)
+   3. [Building on GNU/Linux](#3-building-on-gnulinux)
+   4. [Building on Windows](#3-building-on-windows)
```
Comment (Codym48) [inline INSTALL.md:7]:
```suggestion
   4. [Building on Windows](#4-building-on-windows)
```

File: INSTALL.md [hunk 2]
```diff
@@ -0,0 +1,1554 @@
...
+   5. [Building on MacOS X](#4-building-on-macos-x)
```
Comment (Codym48) [inline INSTALL.md:8]:
```suggestion
   5. [Building on MacOS X](#5-building-on-macos-x)
```

File: INSTALL.md [hunk 3]
```diff
@@ -0,0 +1,1554 @@
...
+  After the build completed you should install QGIS using the INSTALL target.
+
+  Install QGIS by building the INSTALL project. By default this will install to
+  c:\Program Files\qgis<version> (this can be changed by changing the
+  CMAKE_INSTALL_PREFIX variable in cmake-gui).
+
+  You will also either need to add all the dependency DLLs to the QGIS install
+  directory or add their respective directories to your PATH.
+
+### 4.1.5. Packaging
```
Comment (Codym48) [inline INSTALL.md:605]:
```suggestion
### 4.1.6. Packaging
```

File: INSTALL.md [hunk 4]
```diff
@@ -0,0 +1,1554 @@
...
+NSIS for Windows can be downloaded at:
+
+> http://nsis.sourceforge.net
+
+And Perl for Windows (including other requirements like 'wget', 'unzip', 'tar'
+and 'bzip2') is available at:
+
+> http://cygwin.com
+
+### 4.1.6. Packaging your own build of QGIS
```
Comment (Codym48) [inline INSTALL.md:624]:
```suggestion
### 4.1.7. Packaging your own build of QGIS
```

File: INSTALL.md [hunk 5]
```diff
@@ -0,0 +1,1554 @@
...
+Now create a package.
+
+```cmd
+    ./quickpackage.sh
+```
+
+After this you should now have a nsis installer containing your own build
+of QGIS and all dependencies needed to run it on a windows machine.
+
+### 4.1.7. Osgeo4w packaging
```
Comment (Codym48) [inline INSTALL.md:646]:
```suggestion
### 4.1.8. Osgeo4w packaging
```

File: INSTALL.md [hunk 6]
```diff
@@ -0,0 +1,1554 @@
...
+a Cygwin[64] Terminal
+
+And, to avoid Git in Windows reporting changes to files not actually modified:
+
+```cmd
+    cd QGIS
+    git config core.filemode false
+```
+
+### 4.1.4. Setting up the Visual Studio project with CMake
```
Comment (Codym48) [inline INSTALL.md:519]:
```suggestion
### 4.1.4. Configure and build with CMake from command line
```

File: INSTALL.md [hunk 7]
```diff
@@ -0,0 +1,1554 @@
...
+  We will need to run MSVC with all the environment variables set, thus we will run it as follows:
+  Run the batch file OSGeo4W-dev.bat you created before.
+  On the command prompt run `call gdal-dev-env.bat` to add the release gdal and proj libraries to your PATH.
+  On the command prompt run `devenv` to open MSVC.
+  From MSVC, open the solution file
+  `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\qgis.sln`
+  Try to build the solution (go grab a cup of tea, it may take a while).
+  If it fails, run it again and again until there are (hopefully) no errors.
+
+Running QGIS from within MSVC:
```
[the remark this answers is not in the document]
Comment (Codym48) [reply INSTALL.md:559]: I think a header section here would be worthwhile for consistency with 4.1.4.1 and 4.1.4.2

File: INSTALL.md [hunk 8]
```diff
@@ -0,0 +1,1554 @@
...
+    configonly.bat
+```
+
+#### 4.1.4.2 Compiling QGIS with MSVC:
+  We will need to run MSVC with all the environment variables set, thus we will run it as follows:
+  Run the batch file OSGeo4W-dev.bat you created before.
+  On the command prompt run `call gdal-dev-env.bat` to add the release gdal and proj libraries to your PATH.
+  On the command prompt run `devenv` to open MSVC.
+  From MSVC, open the solution file
+  `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\qgis.sln`
```
Comment (Codym48) [inline INSTALL.md:555]:
```suggestion
  `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\qgis.sln`.
```

File: INSTALL.md [hunk 9]
```diff
@@ -0,0 +1,1554 @@
...
+  Try to build the solution (go grab a cup of tea, it may take a while).
+  If it fails, run it again and again until there are (hopefully) no errors.
+
+Running QGIS from within MSVC:
+  Edit the properties of the project ALL_BUILD:
```
Comment (Codym48) [inline INSTALL.md:560]:
```suggestion
  Edit the properties of the project ALL_BUILD to include the path to the executable:
```

File: INSTALL.md [hunk 10]
```diff
@@ -0,0 +1,1554 @@
...
+  Debugging -> Command -> `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\output\bin\RelWithDebInfo\qgis.exe`
+  Run: Debug -> Start Debugging (F5) or Start Without Debugging (Ctrl+F5)
```
Comment (Codym48) [inline INSTALL.md:562]:
```suggestion
  To run, use the menu commands: Debug -> Start Debugging (F5) or Start Without Debugging (Ctrl+F5).
```

File: INSTALL.md [hunk 11]
```diff
@@ -0,0 +1,1554 @@
...
+  On the command prompt run `call gdal-dev-env.bat` to add the release gdal and proj libraries to your PATH.
+  On the command prompt run `devenv` to open MSVC.
+  From MSVC, open the solution file
+  `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\qgis.sln`
+  Try to build the solution (go grab a cup of tea, it may take a while).
+  If it fails, run it again and again until there are (hopefully) no errors.
+
+Running QGIS from within MSVC:
+  Edit the properties of the project ALL_BUILD:
+  Debugging -> Command -> `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\output\bin\RelWithDebInfo\qgis.exe`
```
Comment (Codym48) [inline INSTALL.md:561]:
```suggestion
  Debugging -> Command -> `C:\OSGeo4W64\QGIS\ms-windows\osgeo4w\build-qgis-test-x86_64\output\bin\RelWithDebInfo\qgis.exe`.
```

**Review by Codym48 — CHANGES_REQUESTED** [review]

**Comment by Codym48** [comment]
Could also update the **Building from Source** section [at the bottom of the README.md](https://github.com/PeterPetrik/QGIS/blob/remove_txt2txt/README.md#building-from-source) to point to the new INSTALL.md? (It currently references doc/INSTALL.html which won't exist anymore.)

Could also update the text or links in this **Note** [in the README.md](https://github.com/PeterPetrik/QGIS/blob/remove_txt2txt/README.md#note) that mentions the "installation documentation," but actually refers to the [overall documentation webpage](https://qgis.org/en/docs/index.html), and mentions a PDF file in the doc directory that doesn't exist.

**Comment by PeterPetrik** [comment]
@Codym48 thanks for review. 

At this point I would like to merge this somehow to master (after @jef-n maybe can verify/fix the perl scripts), so such improvement to the text could be done in separate pull requests...

**Review by Codym48 — APPROVED** [review]

**Comment by nyalldawson** [comment]
Lets go ahead and merge -- it'll be easier to identify any issues in the scripts post merge...

**Pull request merged**